libparsec/crates/client/src/client/mod.rs

```diff
@@ -533,8 +541,192 @@ impl Client {
     pub async fn accept_tos(&self, tos_updated_on: DateTime) -> Result<(), ClientAcceptTosError> {
         tos::accept_tos(self, tos_updated_on).await
     }
+
+    pub async fn export_recovery_device(
+        &self,
+        device_label: DeviceLabel,
+    ) -> Result<(SecretKeyPassphrase, Vec<u8>), ExportRecoveryDeviceError> {
+        let (passphrase, data, recovery_device) =
+            libparsec_platform_device_loader::inner_export_recovery_device(
+                &self.device,
+                device_label,
+            )
+            .await?;
+
+        // save recovery device
+        let outcome =
+            create_new_device(self.cmds.clone(), recovery_device, self.device.clone()).await?;
+
+        let latest_known_timestamps = match outcome {
+            CertificateBasedActionOutcome::LocalIdempotent => return Ok((passphrase, data)), // not sure how it could happen though
+            CertificateBasedActionOutcome::Uploaded {
+                certificate_timestamp,
+            }
+            | CertificateBasedActionOutcome::RemoteIdempotent {
+                certificate_timestamp,
+            } => PerTopicLastTimestamps::new_for_common_and_shamir(
+                certificate_timestamp,
+                certificate_timestamp,
+            ),
+        };
+        self.certificates_ops
+            .poll_server_for_new_certificates(Some(&latest_known_timestamps))
+            .await
+            .map_err(|e| match e {
+                CertifPollServerError::Stopped => ExportRecoveryDeviceError::Stopped,
+                CertifPollServerError::Offline => ExportRecoveryDeviceError::Offline,
+                CertifPollServerError::InvalidCertificate(err) => {
+                    ExportRecoveryDeviceError::InvalidCertificate(err)
+                }
+                CertifPollServerError::Internal(err) => err
+                    .context("Cannot poll server for new certificates")
+                    .into(),
+            })?;
+        Ok((passphrase, data))
+    }
+}
+pub async fn create_device_from_recovery(
+    cmds: Arc<AuthenticatedCmds>,
+    recovery_device: &LocalDevice,
+    device_label: &DeviceLabel,
+    save_strategy: DeviceSaveStrategy,
+    config_dir: PathBuf,
+) -> Result<AvailableDevice, ImportRecoveryDeviceError> {
+    let new_device = LocalDevice::generate_new_device(
```

Comment:
Let's add a new method `LocalDevice::new_device_from_existing_device` so we don't run into more problems in the future

Reply:
:+1: 

I would like if we go with `from_existing_device`, for me the prefix `new_device` is already implied with the `from`.